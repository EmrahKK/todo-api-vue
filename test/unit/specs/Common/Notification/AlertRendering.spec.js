import Vue from 'vue'
import Vuex from 'vuex'
import { mount } from 'vue-test-utils'
import Alert from '@/components/Common/Notification/Alert'
import alertModule from '@/store/modules/alert'

Vue.use(Vuex)

describe('Alert component rendering:', () => {
  let getters
  let store

  beforeEach(() => {
    getters = {
      status: alertModule.getters.status
    }
  })

  it('Should render a hidden alert', () => {
    store = new Vuex.Store({
      getters,
      state: {
        status: {
          cssClass: 'alert-info',
          message: 'Some alert message',
          showing: false
        }
      }
    })

    const wrapper = mount(Alert, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Should render a visible alert', () => {
    store = new Vuex.Store({
      getters,
      state: {
        status: {
          cssClass: 'alert-info',
          message: 'Some alert message',
          showing: true
        }
      }
    })

    const wrapper = mount(Alert, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })
})
