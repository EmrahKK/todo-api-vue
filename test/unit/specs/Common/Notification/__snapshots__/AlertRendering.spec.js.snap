// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component rendering: Should render a hidden alert 1`] = `
<div class="statusElement alert alert-info">
    Some alert message
</div>
`;

exports[`Alert component rendering: Should render a visible alert 1`] = `
<div class="statusElement alert alert-info showing">
    Some alert message
</div>
`;
